src/main/java/duke/models/tasks/TaskManager.java

```diff
@@ -60,18 +60,30 @@ public TaskManager() {
         return tasksWithThisDescription;
     }
 
-
     /**
      * Adds a Task to the list.
      *
-     * @param task The Task to be added to the list.
+     * @param taskToBeAdded contains the Task to be added to the list.
+     * @throws DukeException if the task is not added successfully.
      */
-    public void addTask(Task task) {
-        if (task.getId() == 0) {
+    public void addTask(Task taskToBeAdded) throws DukeException {
+        if (taskToBeAdded.getDescription().length() < 4) {
+            throw new DukeException("The task description is too short.");
+        }
+        ArrayList<Task> sameTasks = new ArrayList<>();
+        for (Task task : taskIdMap.values()) {
+            if (task.getDescription().toLowerCase().equals(taskToBeAdded.getDescription())) {
```

Comment:
Should probably take the `taskToBeAdded.getDescription()` and make sure it's lowercase, too. Otherwise string matches might get overlooked due to case sensitivity.

Reply:
You are right. I will add the toLowerCase on taskToBeAdded.getDescription as well.